src/System.Net.Security/tests/Scripts/setup-kdc.sh

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env bash
+
+OS=`cat /etc/os-release | grep "^ID=" | sed 's/ID=//g' | sed 's/["]//g' | awk '{print $1}'`
+echo "Operating System: ${OS}"
+
+realm="TEST.COREFX.NET"
+
+principal1="TESTHOST/testfqdn.test.corefx.net"
+principal2="TESTHTTP"
+krb_user="krb_user"
+password="password"
+
+kadmin="kadmin.local"
+krb5kdc="krb5kdc"
+kdb5_util="kdb5_util"
+add_principal_cmd="add_principal -pw ${password}"
+
+krb_conf="krb5.conf"
+krb_conf_location="/etc/krb5.conf"
+keytabfile="/etc/krb5.keytab"
+
+PROGNAME=$(basename $0)
+usage()
+{
+	echo "Usage: ${PROGNAME} [-y|--yes] [-u|--uninstall]";
+}
+
+# Cleanup config files and uninstall KDC
+clean_up()
+{
+	echo "Stopping KDC.."
+	if pgrep krb5kdc 2> /dev/null; then sudo killall krb5kdc ; fi
+	
+	echo "Removing config files"
+	if [ -f ${krb_conf_location} ]; then
+		sudo bash -c "rm -f ${krb_conf_location}"
+	fi
+	
+	case ${OS} in
+		"ubuntu" | "debian")
+			kdc_conf_location="/etc/krb5kdc/kdc.conf"
+			dpkg -s krb5-kdc >/dev/null 2>&1
+			if [ $? -eq 0 ]; then
+				echo "Uninstalling krb5-kdc"
+				sudo apt-get -y purge krb5-kdc
+			fi
+		;;
+		"centos" | "rhel")
+			kdc_conf_location="/var/kerberos/krb5kdc/kdc.conf"
+			yum list installed krb5-server >/dev/null 2>&1
+			if [ $? -eq 0 ]; then
+				echo "Uninstalling krb5-server"
+				sudo yum -y remove krb5-server
+			fi
+		;;
+
+		*)
+		;;
+	esac
+	
+	if [ -f ${kdc_conf_location} ]; then
+		sudo bash -c "rm -f ${kdc_conf_location}"
+	fi
+}
+
+error_exit()
+{
+	echo "${1:-"Unknown Error"}"
+	echo "Aborting"
+	clean_up
+	exit 1
+}
+
+# Common function across linux distro's to configure KDC post installation
+configure_kdc()
+{
+	echo "Stopping KDC.."
+	if pgrep krb5kdc 2> /dev/null; then sudo killall krb5kdc ; fi
+	
+	# Remove database files if exist
+	# There are multiple kdc db files with name pattern principal*. Doing "sudo rm -f" doesn't work,
+	# because the file name expansion happens in context of normal user, (who does not have pemission on the path)
+	# and then the command is passed to sudo. Hence it tries to delete file with name "principal*", which does not exist.
+	# To overcome this and make file name expansion under sudoer/superuser, rm is run under a separate bash
+	sudo bash -c "rm -f ${database_files}"
+	
+	# Create/copy krb5.conf and kdc.conf
+	echo "Copying krb5.conf and kdc.conf.."
+	sudo /bin/cp ${krb_conf} ${krb_conf_location} || \
+	error_exit "Cannot copy ${krb_conf} to ${krb_conf_location}"
+	
+	sudo /bin/cp ${kdc_conf} ${kdc_conf_location} || \
+	error_exit "Cannot copy ${kdc_conf} to ${kdc_conf_location}"
```

Comment:
What's the state of the machine if one of these errors out?  Will uninstall successfully undo any changes made thus far, even if the install wasn't complete?  I just want to make sure that a dev can get back to a clean state if something goes wrong part way through install.